src/Parser.php

```diff
@@ -218,12 +224,28 @@ protected function parseFeature()
         while ($this->predictTokenType() !== 'EOS') {
             $node = $this->parseExpression();
 
-            if (is_string($node)) {
+            if (is_string($node) && $this->compatibilityMode->shouldRemoveFeatureDescriptionPadding()) {
                 $text = preg_replace('/^\s{0,' . ($token['indent'] + 2) . '}|\s*$/', '', $node);
                 $description .= ($description !== null ? "\n" : '') . $text;
                 continue;
             }
 
+            if (is_string($node)) {
```

Comment:
The code inside this if statement only runs if we are not in legacy compatibility mode but I had to think twice to realize this. I think the logic would be clearer if we did:

```
if (is_string($node)) {
    if ($this->compatibilityMode->shouldRemoveFeatureDescriptionPadding()) {
        ...
        continue;
   }     
   if ($node === "\n" && $description === null) {
       ...
   }
   ....
   continue;
}
```